rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp

```diff
@@ -72,7 +82,7 @@ void COCOMetaDataReader::lookup(const std::vector<std::string> &image_names)
     }
 }
 
-void COCOMetaDataReader::add(std::string image_name, BoundingBoxCords bb_coords, Labels bb_labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count)
+void COCOMetaDataReader::add(std::string image_name, BoundingBoxCords bb_coords, Labels bb_labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count, int image_id)
```

Comment:
Can this image_id be -ve ?
If not we can used unsigned int for image_id

Reply:
Image ID is always negative. When we read from json it is an int value not unsigned hence maintaining it as int.
In open source also it was int only